### ISSUE TITLE
[ENH] Different exogenous features for different time series using a global model

### ISSUE THREAD

**Issue by ncooder** [sktime/sktime#7484]
Is there a way to use different exogenous features for different time series within a global model? For example, I would like to use feature_1 and feature_2 for time series ID_1, while using feature_3 and feature_4 for time series ID_2, and so on. Simply filling unused features with zeros for each time series ID is not a good idea. Does sktime have an existing solution for this scenario?

**Comment by yarnabrina** [comment]
I don't think we support this for the time being. Inviting @benHeid @XinyuWuu for their inputs too.

Can you please tell some real life use cases for the same? For example if different series (as identified by hierarchy or panel) are affected by different features, is it not more likely that different forecasters for them to be better?

**Comment by ncooder** [comment]
@yarnabrina It is about the impact of marketing for different channels like social media features, advertisement spending etc. where each product sales may depend more on different features also in different places.

**Comment by yarnabrina** [comment]
I can totally see different values of the same external features, e.g. different discount rates (which may be 0 if campaign is not going everywhere) or different holidays or different climates and etc. but I am unable to quickly think of a feature variable that affects in one place and in other place does not affect at all. Can you give a specific example please?

I am quite confident that `sktime` global models do not support this, as even the general non-global models also won't support this. But with more info, we can consider this as a possible future project and plan on supporting it.

**Comment by julian-fong** [comment]
Don't most if not all (classical) machine learning models depend on the correct amount of predictors passed in that was originally used to train the model (i.e passing in 7 columns in a model that was trained on 6 columns will not work)? Or am I mis-understanding the scenario here

**Comment by fkiraly** [comment]
Classical models and reduction models typically yes, and in fact all models in `sktime` currently make this assumption.

There is an mtype supporting different features per instance (`df-list`), but we have never used it because we have never encountered a model that could make use of this.

Perhaps you could give some example models that we could interface and test this for, @ncooder?

**Comment by benHeid** [comment]
I think this is a very nice request. Foundation Models that support exogenous features should be able to handle it. Otherwise, I can imagine the following approaches that could support it in theory:
* Forecasting network with multiple encoders (One encoder per feature set which can be exchanged depending on the incoming time series) but only one prediction head. 
More mathematical: We have two encoder `enc1` and `enc2` and one prediction head `pred`, depending on the input time series either `pred(enc1(y))` or `pred(enc2(y))` is applied. I.e., during `fit` we would need the information which encoder should be applied for which time series in the hierarchical data or in the panel. The same information, we would also require during `predict`. Not sure if this is derivable from the input. 
* A masking approach. I.e., in theory the network is supporting up to `n` exogenous features. These features can be masked if they are not provided for a specific time series. This, would require that masking information is also passed to `fit` and `predict`.

Not sure, if there are more approaches to tackle that problem. But at least the upper two seems to be solvable by extending `fit` and `predict` with the possibility to pass masks.
